### ISSUE TITLE
Collapsible JSON logs

### ISSUE THREAD

**Issue by owenchak** [tilt-dev/tilt#5993]
## Describe the Feature You Want
Logs are often in JSON format containing a message field and any other metadata such as a stack trace, service name, etc. Often times, I only care about the message and all this additional metadata makes it hard to parse the logs. In the web mode, it would be great to have a configuration option to specify the log format (eg. JSON, plain text, etc) and if the format is JSON, have another option to collapse logs by default. When logs are collapsed, only the message field is shown, otherwise, display the entire JSON object. Wanted to get a gut check on this feature request before I spend time putting up a PR. Thank you!

### Current Behavior
The entire JSON log is rendered.

### Why Do You Want This?
There is too much additional metadata in logs which makes it hard to parse logs.

**Additional context**
N/A

**Comment by nicks** [comment]
thanks for the feature request! I'm going to mark this as a duplicate of https://github.com/tilt-dev/tilt/issues/4581

but yeah, i think this has been requested a few times. we would accept a PR for it, but probably good to post the syntax you have in mind before you start working on it.
